linode/rdns/framework_resource.go

```diff
@@ -39,236 +31,126 @@ type Resource struct {
 	helper.BaseResource
 }
 
-func (r *Resource) Create(
-	ctx context.Context,
-	req resource.CreateRequest,
-	resp *resource.CreateResponse,
-) {
+func (r *Resource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
 	tflog.Debug(ctx, "Create linode_rdns")
 
 	var plan ResourceModel
-	client := r.Meta.Client
-
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	ctx = populateLogAttributes(ctx, plan)
+	client := r.Meta.Client
+	updateOpts := linodego.IPAddressUpdateOptions{}
 
-	createTimeout, diags := plan.Timeouts.Create(ctx, DefaultVolumeCreateTimeout)
-	resp.Diagnostics.Append(diags...)
-	if resp.Diagnostics.HasError() {
-		return
+	if !plan.RDNS.IsNull() {
+		updateOpts.RDNS = plan.RDNS.ValueStringPointer()
 	}
 
-	ctx, cancel := context.WithTimeout(ctx, createTimeout)
-	defer cancel()
-
-	address := plan.Address.ValueString()
-
-	ip, err := client.GetIPAddress(ctx, address)
-	if err != nil {
-		resp.Diagnostics.AddError(
-			"Failed to get the ip address associated with this RDNS",
-			err.Error(),
-		)
-		return
+	if !plan.Reserved.IsNull() {
+		reserved := plan.Reserved.ValueBool()
```

Comment:
As we discussed offline, it does not make sense to make `reserved` a property of the `linode_rdns` resource, because it is not related to RDNS. `reserved` is a property of an IP address and show maybe we need to create a new resource that let's user reserve or unreserve an address assigned to a Linode.

Reply:
> As we discussed offline, it does not make sense to make reserved a property of the linode_rdns resource

Agreed, I think it might make more sense for an IP's ephemeral vs reserved status to only be manageable through the IP resource(s) to reduce usage complexity. I'd definitely be interested to hear everyone else's thoughts though!